app/services/confirm_enrolment.rb

```diff
@@ -1,13 +1,19 @@
 class ConfirmEnrolment
-  Response = Struct.new(:successful?, :application_choice)
+  attr_reader :errors, :application_choice
 
   def initialize(application_choice:)
     @application_choice = application_choice
   end
 
-  def call
-    ApplicationStateChange.new(@application_choice).confirm_enrolment!
-
-    Response.new(true, @application_choice)
+  def save
+    ApplicationStateChange.new(application_choice).confirm_enrolment!
+    true
+  rescue Workflow::NoTransitionAllowed => e
```

Comment:
Ought we to include `ActiveModel::Validations` in these objects? Then we get error key-value matching for free, plus `.full_sentences` etc.